servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1320,38 +1305,88 @@ public long loadMetadata(TachyonURI path, boolean recursive)
     }
   }
 
-  public boolean mount(TachyonURI tachyonPath, TachyonURI ufsPath)
-      throws FileAlreadyExistsException, FileDoesNotExistException, InvalidPathException,
-      IOException {
-    synchronized (mInodeTree) {
-      MkdirOptions options =
-          new MkdirOptions.Builder(MasterContext.getConf()).setPersisted(true).build();
-      InodeTree.CreatePathResult createResult = mkdir(tachyonPath, options);
-      if (mountInternal(tachyonPath, ufsPath)) {
-        AddMountPointEntry addMountPoint = AddMountPointEntry.newBuilder()
-            .setTachyonPath(tachyonPath.toString())
-            .setUfsPath(ufsPath.toString())
-            .build();
-        writeJournalEntry(JournalEntry.newBuilder().setAddMountPoint(addMountPoint).build());
-        flushJournal();
-        return true;
+  /**
+   * Loads metadata for the directory identified by the given path from UFS into Tachyon. This does
+   * not actually require looking at the UFS path.
+   *
+   * @param path the path for which metadata should be loaded
+   * @param recursive whether parent directories should be created if they do not already exist
+   * @return the file id of the loaded directory
+   * @throws FileAlreadyExistsException if the object to be loaded already exists
+   * @throws InvalidPathException if invalid path is encountered
+   * @throws IOException if an I/O error occurs
+   */
+  private long loadMetadataDirectory(TachyonURI path, boolean recursive)
+      throws IOException, FileAlreadyExistsException, InvalidPathException {
+    MkdirOptions options =
+        new MkdirOptions.Builder(MasterContext.getConf()).setRecursive(recursive)
+            .setPersisted(true).build();
+    InodeTree.CreatePathResult result = mkdir(path, options);
+    List<Inode> inodes = null;
+    if (result.getCreated().size() > 0) {
+      inodes = result.getCreated();
+    } else if (result.getPersisted().size() > 0) {
+      inodes = result.getPersisted();
+    } else if (result.getModified().size() > 0) {
+      inodes = result.getModified();
+    }
+    if (inodes == null) {
+      throw new FileAlreadyExistsException(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path));
+    }
+    return inodes.get(inodes.size() - 1).getId();
+  }
+
+  public synchronized boolean mount(TachyonURI tachyonPath, TachyonURI ufsPath)
```

Comment:
Why is this method synchronized? Shouldn't the body be synchronized on mInodeTree?